common/djangoapps/track/contexts.py

```diff
@@ -106,3 +115,23 @@ def context_dict_for_learning_context(context_key):
         if hasattr(context_key, 'enterprise_uuid'):
             context_dict['enterprise_uuid'] = context_key.enterprise_uuid
     return context_dict
+
+
+def enterprise_context_from_course_id_user_id(course_id, user_id):
+    """
+    Creates an enterprise context from a `course_id` anf `user_id`.
+
+    Example Returned Context::
+
+        {
+            'enterprise_uuid': '1a0fbcbe-49e5-42f1-8e83-4cddfa592f22'
+        }
+    """
+    context_dict = {}
+    if ENTERPRISE_ENABLED and enterprise_enabled() and is_enterprise_learner(user_id):
```

Comment:
@rehanedly only these two condition are enough to execute enterprise related logic
`enterprise_enabled() and is_enterprise_learner(user_id)`

Reply:
fixed